src/controller/CHIPDeviceController.cpp

```diff
@@ -1771,12 +1771,12 @@ void DeviceCommissioner::CleanupCommissioning(DeviceProxy * proxy, NodeId nodeId
     if (completionStatus.err == CHIP_NO_ERROR)
     {
         CommissioneeDeviceProxy * commissionee = FindCommissioneeDevice(nodeId);
+        CommissioningStageComplete(CHIP_NO_ERROR);
```

Comment:
OK, I see.  So in the "normal" path `commissionee` and `mDeviceBeingCommissioned` are not the same object.

So then this makes some sense, except for the following: when we call `CommissioningStageComplete` that will send client notifications, and those can cause `commissionee` to be destroyed, no?  And then we will be doing use-after-free when we go to release it.

Reply:
@bzbarsky-apple 
No, there is no such possibility. In the case of the CleanUp stage in CommissioningStageComplete, a code has already been added to skip the release.

https://github.com/project-chip/connectedhomeip/blob/3cb83827fc6d46d66fc34ca0a39b54dff2170c00/src/controller/CHIPDeviceController.cpp#L1932

Reply:
Hmm, ok.  In theory, `CommissioningStageComplete` can tear down the controller, but that has other problems too.  Alright, thanks!